pkg/downloader/chart_downloader.go

```diff
@@ -365,34 +368,70 @@ func pickChartRepositoryConfigByName(name string, cfgs []*repo.Entry) (*repo.Ent
 // Charts are not required to be included in an index before they are valid. So
 // be mindful of this case.
 //
-// The same URL can technically exist in two or more repositories. This algorithm
-// will return the first one it finds. Order is determined by the order of repositories
-// in the repositories.yaml file.
+// The same URL can technically exist in multiple repositories.
+// This algorithm will return one of them based on concurrent processing,
```

Comment:
I do think we need to be careful here. Helm is used in many, many places. And small brhavioral changes like this can affect folk

ie. someone production pipeline is working, because their `repositories.yaml` happens to have the correct repo for their environment first. This change (based on the non-deterministic order of scheduling go-routines) might return a different result.

Reply:
I think (also) worth asking in #helm-dev or on the helm developer call whether this behavior can/should be changed

Reply:
@gjenkins8, but the helm doesn't document this behavior and doesn't guarantee it.